src/test/java/com/CarRentalAgency/controller/CarControllerTest.java

```diff
@@ -0,0 +1,148 @@
+package com.CarRentalAgency.controller;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.when;
+
+import com.CarRentalAgency.entity.Car;
+import com.CarRentalAgency.services.CarService;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+@ExtendWith(MockitoExtension.class)
+public class CarControllerTest {
+
+  @Mock
+  private CarService carService;
+
+  @InjectMocks
+  private CarController underTest;
+
+
+  @Test
+  public void testSaveCar() {
+    Car car = new Car();
+    car.setModel(Car.Model.SEDAN);
+    car.setRegistrationNumber(1234);
+    car.setKilometre(1000);
+    car.setCarName("test");
+    ResponseEntity<Car> responseEntity = underTest.saveCar(car);
+    assertEquals(HttpStatus.CREATED, responseEntity.getStatusCode());
```

Comment:
**Suggestion:** Use a setup method to avoid code duplication in your tests.
```suggestion
  private Car car;

  @BeforeEach
  public void setup() {
    car = new Car();
    car.setModel(Car.Model.SEDAN);
    car.setRegistrationNumber(1234);
    car.setKilometres(1000);
    car.setName("Test Car");
  }

  public void testSaveCar() {
    ResponseEntity<Car> responseEntity = underTest.saveCar(car);
    assertEquals(HttpStatus.CREATED, responseEntity.getStatusCode());
  }
```